custom_components/econnect_metronet/binary_sensor.py

```diff
@@ -131,21 +131,58 @@ def name(self) -> str:
     @property
     def icon(self) -> str:
         """Return the icon used by this entity."""
-        if self._sensor_type == query.SECTORS:
-            return "hass:shield-home-outline"
-        elif self._sensor_type == query.INPUTS:
-            return "hass:electric-switch"
-        else:
-            return "hass:toggle-switch-off-outline"
+        return "hass:electric-switch"
 
     @property
     def is_on(self) -> bool:
         """Return the binary sensor status (on/off)."""
         # TODO: evaluate to expose a device API to get the status of a sensor
         # instead of duplicating this sensor_type check
-        if self._sensor_type == query.SECTORS:
-            return self._device.sectors_armed.get(self._sensor_id) is not None
-        elif self._sensor_type == query.INPUTS:
-            return self._device.inputs_alerted.get(self._sensor_id) is not None
-        else:
-            return False
+        return self._device.inputs_alerted.get(self._sensor_id) is not None
+
+
+class SectorSensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a e-connect sector binary sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        unique_id: str,
+        sector_id: int,
+        config: ConfigEntry,
+        name: str,
+        coordinator: DataUpdateCoordinator,
+        device: AlarmDevice,
+        sensor_type: int,
+    ) -> None:
+        """Construct."""
+        super().__init__(coordinator)
+        self.entity_id = generate_entity_id(config, name)
+        self._name = name
+        self._device = device
+        self._unique_id = unique_id
+        self._sector_id = sector_id
+        self._sensor_type = sensor_type
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique identifier."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of this entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
```

Comment:
For all classes, I would add one test that verifies if `hass:shield-home-outline` is what you get from `entity.icon`. This sets in stone the value, meaning if by any chance we're changing this code, the test fails if the icon is not what we got.